cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -544,41 +529,55 @@ impl ManagedEnvironment {
     }
 }
 
-/// Write a pointer lockfile to the specified `lock_path` storing the
-/// current git revision of the tracked upstream repository identified by a `pointer`.
+/// Write a pointer lockfile to the specified `lock_path`.
+///
+/// The lockfile stores the current git revision of the tracked upstream repository.
+/// When a local revision is specified,
+/// and the local revision is different from the remote revision,
+/// the local revision is also stored in the lockfile.
 ///
 /// When committed to a project, guarantees
 /// that the same version of the linked environment is used by all clones.
+/// When a local revision is specified,
+/// flox will **try to** check out the local revision
```

Comment:
blocking: I don't understand what you mean by check out, can you explain that?

Reply:
It's supposed to refer to `git checkout`
Not quite sure what's a better term here

Reply:
Oh I just misunderstood you were talking about once the lockfile got pulled onto another machine, sorry